File: solutions/go/difference-of-squares/1/difference_of_squares.go
package diffsquares

func SquareOfSums(n int) int {
	return (n * (n + 1) / 2) * (n * (n + 1) / 2)
}

func SumOfSquares(n int) int {
	return n * (n + 1) * (2*n + 1) / 6
}

func Difference(number int) int {
	return SquareOfSums(number) - SumOfSquares(number)
}

File: solutions/ruby/difference-of-squares/1/difference_of_squares.rb
class Squares
  def initialize(number)
    @number = number
  end

  def sum_of_squares
    (1..@number).inject(0) { |sum, n| sum + n**2 }
  end

  def square_of_sum
    (1..@number).sum**2
  end

  def difference
    square_of_sum - sum_of_squares
  end
end